cli/cmd/upgrade.go

```diff
@@ -419,6 +453,18 @@ func (options *upgradeOptions) fetchIdentityValues(k kubernetes.Interface, idctx
 		return nil, err
 	}
 
+	cred, err := issuerData.VerifyAndBuildCreds("")
```

Comment:
I think this can fail for more than one reason. Not necessarily because the verification logic faile.d It might be that the issuer cert does not have the correct crypo requirements and we need to catch that as well. I have changed the error to be of the following format:
```
× Failed to build upgrade configuration: issuer certificate does not work with the provided provided roots: x509: certificate signed by unknown authority
For more information: https://linkerd.io/2/tasks/rotating_identity_trust_root/
For troubleshooting help, visit: https://linkerd.io/upgrade/#troubleshooting
```

Would oyu say that works?

Reply:
Yup!  Having that link to the cert rotation docs is the most important thing.